### DISCUSSION TITLE
Training datasets with variable number of sites

### DISCUSSION THREAD

**Discussion by jacksund** [deepmodeling/deepmd-kit#1140] [Q&A] [answered]
Hello!

How would I train a model using multiple datasets of different type.raw structures?

For example, say I have two datasets -- which are both Carbon structures but each has a different number of atoms. The first dataset is for structures with 4 atoms in each structure. The second dataset is for 8 atoms in each structure.

This is my current folder organization:
```
data
    carbon_04_train
        set.000
        type.raw
        type_map.raw
    carbon_04_test
    carbon_08_train
    carbon_08_test
input.json
```

Where the input.json is... (I removed _comment fields  for clarity)
```
{
    "model": {
	"descriptor" :{
	    "type":		"se_e2_a",
	},
	"fitting_net" : {},
    },
    "learning_rate" :{},
    "loss" :{},
    "_comment": " traing controls",
    "training" : {
	"training_data": {
	    "systems":		["../data/carbon_04_train", "../data/carbon_08_train"],
	},
	"validation_data":{
	    "systems":		["../data/carbon_04_test", "../data/carbon_08_test"],
	},
	"numb_steps":	1000000,
    },
}
```

Is this possible with DeePMD? Thanks in advance,
-Jack

**Comment by njzjz** [comment, marked as answer]
Yes, you can do it. We have given [an example](https://github.com/deepmodeling/deepmd-kit/tree/master/examples/nopbc) in the `examples` directory.

**Reply by jacksund** [reply]
Awesome, thank you for the quick response. I saw that example earlier, but it's not clear which flag in [that input](https://github.com/deepmodeling/deepmd-kit/blob/master/examples/nopbc/train/input.json) turns flexible composition on. Is it handled by fitting_net?

**Reply by njzjz** [reply]
It's supported by default. Nothing needs to turn on.

**Reply by jacksund** [reply]
Ah gotcha. I have it working now -- my problem ended up being a typo in one of type.raw files. :sweat_smile:
